### PR TITLE
Plotting updates

### LINKED ISSUE #217: Toggle plotting for individual samples
Add dynamic context menu actions for toggling showing the plot for individual samples within the "All samples plot".

### LINKED ISSUE #227: Structure factor plotting options
Currently we have the option to plot mint01 and mdcs01 files together.  We should split this into separate options for mint01 only, mdcs01 only, or mint01/mdcs01 together.  This option should be available for both individual samples as well as the combined plotting panel.

### LINKED ISSUE #230: Hide / show plot series via legend?
`gnuplot` has one nice feature (under some window systems) where clicking on an item in the legend toggles the visbiliity of the related data in the plot, offering a quick and easy way to manage data visibility.

If possible, a similar implementation in our plots would be a nice touch.

### LINKED ISSUE #239: Plot translation
I miss the ability to translate the view in a plot - e.g. by holding the middle mouse button and moving the mouse, I would like to be able to move the view around.

### PR DESCRIPTION
**Pull request by jswift-stfc**
PR to update the plotting side of things.

- Implements plot translation, closes #239.
- Toggle plotting for each sample, closes #217.
- Implements further structure factor plotting options, closes #227.
- Toggle visibility of series' using legend, closes #230.

### CODE REVIEW

**Review by trisyoungs — CHANGES_REQUESTED** [review]
Briefly:

- Translation works nicely.
- Toggling visibility with the legend items is awesome.

But...
- There are no axes on the Sample mdcs01 plot (and no expected DCS level)
- Rubberband zoom still doesn't affect the x axis range (it's not that its clamped to the data range)
- On the main `Plots` tab, only the first Sample is plotted for the mdcs01 view (for mint01 all of them are there)

File: src/gui/widgets/gudpy_charts.py [hunk 1]
```diff
@@ -410,8 +472,90 @@ def plotSample(self, sample):
...
+                if len(self.data.keys()) > 1:
+                    dcsSeries.setName(f"{sample.name} expected level")
+                else:
+                    dcsSeries.setName("Expected level")
+                dcsSeries.append(
+                    [
+                        QPointF(x, y)
+                        for x, y in self.data[sample]["dcs"]
+                    ]
+                )
```
[the remark this answers is not in the document]
Comment (jswift-stfc) [reply src/gui/widgets/gudpy_charts.py:507]: > 

Rubberband zoom still eludes me unfortunately, similarly to the logarithmic axes... However I do realise the bug that is causing the issues with the axes for the mdcs01 plot, and the single Sample being plotted - which I shall rectify now.

**Comment by jswift-stfc** [comment]
Rubberband zoom still eludes me unfortunately, similarly to the logarithmic axes... However I have resolved the other issues.

**Comment by trisyoungs** [comment]
> Rubberband zoom still eludes me unfortunately, similarly to the logarithmic axes... However I have resolved the other issues.

Sorry - I forgot that we talked about this before!  A pox on you, Qt. Worst. Plotting. Ever.

**Comment by jswift-stfc** [comment]
No problem! I fixed the other problems, so I'm going to merge this in now.

**Pull request merged**